gateway/sensor.go

```diff
@@ -162,50 +164,98 @@ func (g *Gateway) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 
 	errCode := C.setUpGateway(C.int(cfg.Bus))
 	if errCode != 0 {
-		return fmt.Errorf("failed to start the gateway %d", errCode)
+		strErr := parseErrorCode(int(errCode))
+		return fmt.Errorf("failed to set up the gateway: %s", strErr)
 	}
 
 	g.started = true
-	g.receivePackets()
+	g.workers.Add(g.receivePackets)
```

Comment:
Do we have a `workers == nil` check before we stop this routine somewhere?

Reply:
yes! in `Close`